pandas/core/indexes/datetimelike.py

```diff
@@ -527,24 +526,23 @@ def _maybe_box_as_values(self, values, **attribs):
         # - sort_values
         return values
 
+    @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
-        if is_object_dtype(dtype):
-            return self._box_values_as_index()
-        elif is_string_dtype(dtype) and not is_categorical_dtype(dtype):
-            return Index(self.format(), name=self.name, dtype=object)
-        elif is_integer_dtype(dtype):
-            # TODO(DatetimeArray): use self._values here.
-            # Can't use ._values currently, because that returns a
-            # DatetimeIndex, which throws us in an infinite loop.
-            return Index(self.values.astype('i8', copy=copy), name=self.name,
-                         dtype='i8')
-        elif (is_datetime_or_timedelta_dtype(dtype) and
-              not is_dtype_equal(self.dtype, dtype)) or is_float_dtype(dtype):
-            # disallow conversion between datetime/timedelta,
-            # and conversions for any datetimelike to float
-            msg = 'Cannot cast {name} to dtype {dtype}'
-            raise TypeError(msg.format(name=type(self).__name__, dtype=dtype))
-        return super(DatetimeIndexOpsMixin, self).astype(dtype, copy=copy)
+        if is_dtype_equal(self.dtype, dtype) and copy is False:
+            # Ensure that self.astype(self.dtype) is self
+            return self
+
+        new_values = self._eadata.astype(dtype, copy=copy)
+
+        # we pass `dtype` to the Index constructor, for cases like
+        #  dtype=object to disable inference. But, DTA.astype ignores
+        #  integer sign and size, so we need to detect that case and
+        #  just choose int64.
+        dtype = pandas_dtype(dtype)
+        if is_integer_dtype(dtype):
+            dtype = np.dtype("int64")
+
+        return Index(new_values, dtype=dtype, name=self.name)
```

Comment:
pass copy?

Reply:
No, that should be handled by the `self._eadata.astype(dtype, copy=copy)` above

Reply:
so then pass copy=False